runtime/lua/vim/lsp/util.lua

```diff
@@ -419,6 +419,11 @@ function M.apply_text_edits(text_edits, bufnr, offset_encoding)
   if not next(text_edits) then
     return
   end
+
+  if not bufnr or bufnr == 0 then
```

Comment:
The method here is intended to be called with a result from a language server for a specific document/ buffer, so I think at this point this should be a concrete buffer number already. Because of that I'd tend to turn this into an error with an assert.

(Also the type annotation doesn't imply nil)